### PR TITLE
[Storage] Support NFS and SmbMultichannel

### PR DESCRIPTION
**Pull request by blueww**
## Description

Design pass review:
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/467
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/646

The features are already in Az.Storage preview module  

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage.Management/File/NewAzureStorageShare.cs [hunk 1]
```diff
@@ -112,11 +112,32 @@ public string AccessTier
 
         [Parameter(Mandatory = false, HelpMessage = "Create a snapshot of existing share with same name.")]
         public SwitchParameter Snapshot { get; set; }
+        
+        [Parameter(Mandatory = false,
+            HelpMessage = "Sets protocols for file shares. It cannot be changed after file share creation. Possible values include: 'SMB', 'NFS'")]
+        [ValidateSet(EnabledProtocols.NFS,
+            EnabledProtocols.SMB,
+            IgnoreCase = true)]
+        public string EnabledProtocol { get; set; }
```
Comment (msJinLei) [inline src/Storage/Storage.Management/File/NewAzureStorageShare.cs:121]: if you very unlikely to extend enabled protocols anymore, it is ok to use validate set. If not, argument completer will be better.

File: src/Storage/Storage.Management/File/NewAzureStorageShare.cs [hunk 2]
```diff
@@ -112,11 +112,32 @@ public string AccessTier
...
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Sets reduction of the access rights for the remote superuser. Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'")]
+        [ValidateSet(RootSquashType.NoRootSquash,
+            RootSquashType.RootSquash,
+            RootSquashType.AllSquash,
+            IgnoreCase = true)]
+        public string RootSquash { get; set; }
```
Comment (msJinLei) [inline src/Storage/Storage.Management/File/NewAzureStorageShare.cs:129]: if you are very unlikely to extend RootSquash types, it is ok to use validate set. If not, argument completer will be better.

File: src/Storage/Storage.Management/ChangeLog.md [hunk 3]
```diff
@@ -18,6 +18,11 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Support create file share with NFS/SMB enabledEnabledProtocol and RootSquash, and update share with RootSquash
```
Comment (msJinLei) [inline src/Storage/Storage.Management/ChangeLog.md:21]: Supported creating .... would be better

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Storage/Storage.Management/File/NewAzureStorageShare.cs (hunk 1 above)
Comment (blueww) [reply src/Storage/Storage.Management/File/NewAzureStorageShare.cs:121]: We don't want user to input value except the NFS/SMB here. It seems argument completer will still allow user to input other values. So it seems ValidateSet will be more suitable here.

BTW, if we add more value to the validationSet, it should not be breaking, right? (I have done such change before and not taken as breaking.)

File: src/Storage/Storage.Management/File/NewAzureStorageShare.cs (hunk 2 above)
Comment (blueww) [reply src/Storage/Storage.Management/File/NewAzureStorageShare.cs:129]: same as above

File: src/Storage/Storage.Management/ChangeLog.md (hunk 3 above)
Comment (blueww) [reply src/Storage/Storage.Management/ChangeLog.md]: fixed.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**